source/Crew.h

```diff
@@ -0,0 +1,64 @@
+/* Crew.h
+Copyright (c) 2019 by Luke Arndt
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef CREW_H_
+#define CREW_H_
+
+#include "Ship.h"
+
+class Crew
+{
+public:
+	// Calculate one day's salaries for the Player's fleet
+	static int64_t CalculateSalaries(const std::vector<std::shared_ptr<Ship>> &ships, const Ship * flagship, const bool includeExtras = true);
+
+	// Calculate the total cost of the flagship's extra crew
+	static int64_t CostOfExtraCrew(const std::vector<std::shared_ptr<Ship>> &ships, const Ship * flagship);
+
+	// Figure out how many of a given crew member are on a ship
+	static int64_t NumberOnShip(const Crew &crew, const std::shared_ptr<Ship> &ship, const bool isFlagship, const bool includeExtras = true);
+
+	// Calculate one day's salaries for a ship
+	static int64_t SalariesForShip(const std::shared_ptr<Ship> &ship, const bool isFlagship, const bool includeExtras = true);
+
+	// Load a definition for a crew member.
+	void Load(const DataNode &node);
+	
+	bool AvoidsEscorts() const;
+	bool AvoidsFlagship() const;
+	int64_t MinimumPerShip() const;
+	int64_t ParkedSalary() const;
+	int64_t PopulationPerMember() const;
+	int64_t Salary() const;
+	const std::string &Id() const;
+	const std::string &Name() const;
+
+private:
+	// If true, the crew member will not appear on escorts
+	bool avoidsEscorts = false;
+	// If true, the crew member will not appear on the flagship
+	bool avoidsFlagship = false;
+	// Each valid ship has at least this many of the crew member
+	int64_t minimumPerShip = 0;
```

Comment:
What if a ship has less crew that the minimum required? Suppose each ship needs a Captain, a Cook and a Pilot. How would that work for ships with a crew of 2?

Reply:
Good point. I've addressed this by replacing "minimum per ship" with a "place at" node. This will allow us to hand-place crew members in specific positions.

This solves the problem because we can have our "minimum per ship" idea with finer-grained control. It also means that we can have non-scaling crew members on bigger ships, such as your Cook.